### PR TITLE
feat(goal_planner): keep margin against objects as possible

### PR DESCRIPTION
**Pull request by kosuke55**
## Description

[feat(goal_planner): keep margin against objects as possible](https://github.com/autowarefoundation/autoware.universe/commit/e73d68329b6c940c7c7b3873dfb25e52f7e73ac8)

before

image

image

https://github.com/autowarefoundation/autoware.universe/assets/39142679/ef539b10-3b4a-4d78-8fef-42838bbf3370

after

image

https://github.com/autowarefoundation/autoware.universe/assets/39142679/a404d9dc-5ebb-46bd-9ba5-591d254eda47

## Related links

tier4 internal ticket
https://tier4.atlassian.net/browse/RT1-4131

## Tests performed

psim

evaluator_description: feat/keep_distance
2023/11/13 https://evaluation.tier4.jp/evaluation/reports/828fceb1-3f15-56bd-b598-2d0be9c3e9ea/?project_id=prd_jt

evaluator_description: feat/keep_distance
2023/11/16 https://evaluation.tier4.jp/evaluation/reports/3feacb18-f7d5-505c-bde4-4d50c98d36f0/?project_id=prd_jt

## Notes for reviewers

none

## Interface changes

none

## Effects on system behavior

none

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp [hunk 1]
```diff
@@ -1329,7 +1400,30 @@ bool GoalPlannerModule::checkCollision(const PathWithLaneId & path) const
     obj_polygons.push_back(tier4_autoware_utils::toPolygon2d(object));
   }
 
-  std::vector<Polygon2d> ego_polygons_expanded;
+  /* Expand ego collision check polygon
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/src/scene_module/goal_planner/goal_planner_module.cpp:1403]: :heart:

File: planning/behavior_path_planner/include/behavior_path_planner/utils/goal_planner/pull_over_planner_base.hpp [hunk 2]
```diff
@@ -53,6 +53,7 @@ struct PullOverPath
   Pose end_pose{};
   std::vector<Pose> debug_poses{};
   size_t goal_id{};
+  size_t id{};
```
Comment (kyoichi-sugahara) [inline planning/behavior_path_planner/include/behavior_path_planner/utils/goal_planner/pull_over_planner_base.hpp:56]: id for path_candidates?

**Review by kyoichi-sugahara — APPROVED** [review]
LGTM!

File: planning/behavior_path_planner/include/behavior_path_planner/utils/goal_planner/pull_over_planner_base.hpp (hunk 2 above)
Comment (kosuke55) [reply planning/behavior_path_planner/include/behavior_path_planner/utils/goal_planner/pull_over_planner_base.hpp:56]: yes

**Comment by kosuke55** [comment]
the actual diff is https://github.com/autowarefoundation/autoware.universe/commit/dec089606b40ab7d639c549c247ce409a8af7b4f

**Pull request merged**